dev-libs/igraph/igraph-0.9.0.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_MAKEFILE_GENERATOR="emake"
+inherit cmake
+
+DESCRIPTION="Creating and manipulating undirected and directed graphs"
+HOMEPAGE="http://www.igraph.org/"
+SRC_URI="https://github.com/igraph/igraph/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0/0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug test threads"
+
+RDEPEND="
+	dev-libs/gmp:0=
+	dev-libs/libxml2
+	sci-libs/arpack
+	sci-libs/cxsparse
+	sci-mathematics/glpk:=
+	virtual/blas
+	virtual/lapack
+"
+DEPEND="${RDEPEND}"
+
+PATCHES=( "${FILESDIR}"/${PN}-0.9.0-cmakedirs.patch )
+
+src_configure() {
+	local mycmakeargs=(
+		-DUSE_CCACHE=OFF
+		-DBUILD_SHARED_LIBS=ON
+		-DIGRAPH_GLPK_SUPPORT=ON
+		-DIGRAPH_GRAPHML_SUPPORT=ON
+		-DIGRAPH_USE_INTERNAL_ARPACK=OFF
+		-DIGRAPH_USE_INTERNAL_BLAS=OFF
+		-DIGRAPH_USE_INTERNAL_CXSPARSE=OFF
+		-DIGRAPH_USE_INTERNAL_GLPK=OFF
+		-DIGRAPH_USE_INTERNAL_GMP=OFF
+		-DIGRAPH_USE_INTERNAL_LAPACK=OFF
+		-DIGRAPH_ENABLE_TLS=$(usex threads)
+		-DBUILD_TESTING=$(usex test)
+	)
+	cmake_src_configure
+}
+
+src_test(){
+	cd "${BUILD_DIR}"
+	emake check
+}
```

Comment:
nope, wrong, `cmake_src_test check`. Does the default test phase function not work?

Reply:
Yes, the default doesn't work, is also why I had to use `emake` as generator.
What you mentioned here, `cmake_src_test check`, also does not work.

Reply:
Using `cmake_src_compile check` in src_test seems to work, would that be more acceptable?

Reply:
yes, use that please

Reply:
and work with upstream on just using CTest so we don't have to override it

Reply:
done

Reply:
fyi, upstream has recently committed a [patch](https://github.com/igraph/igraph/commit/d829842c97cba752c4f1999a832a29ec2c174fbf) that makes it possible to use `make test` similarly to `make check`

Reply:
To be clear, this is not yet in 0.9.1, but it will be in 0.9.2.